src/js/script.js

```diff
@@ -0,0 +1,374 @@
+      let innerKey = document.createElement('div');
+      oneKey.classList.add('outer');
+      if (key.form === 'k') {
+        oneKey.classList.add('key');
+        innerKey.textContent = changeOneKey(key);
+      } else if (key.form === 'p') {
+        oneKey.classList.add('space');
+      } else if (key.form === 'sm') {
+        oneKey.classList.add('small');
+        innerKey.textContent = key.en;
+      } else {
+        oneKey.classList.add('system');
+        innerKey.textContent = key.en;
+      }
+      innerKey.id = key.code;
+      oneKey.append(innerKey);
+      return oneKey;
+    });
+    oneRow.append(...arrayKeys);
+    keyboard.append(oneRow);
+  });
+  let footer = document.createElement('div');
+  footer.classList.add('footer');
+  footer.textContent = 'Windows 10, Change language - Ctrl + Shift';
+  document.body.prepend(field, keyboard, footer);
+  textarea.focus();
+}
+
+function normalizeKeys(code) {
+  if (code === 'Delete') {
+    code = 'NumpadDecimal';
+  } else if (code === 'ArrowLeft') {
+    code = 'Numpad4';
+  } else if (code === 'ArrowRight') {
+    code = 'Numpad6';
+  } else if (code === 'ArrowUp') {
+    code = 'Numpad8';
+  } else if (code === 'ArrowDown') {
+    code = 'Numpad2';
+  }
+  return code;
+}
+
+function keyDown(event) {
+  let code = normalizeKeys(event.code);
+  let element = code && document.getElementById(code);
+  if (element) {
+    element.classList.add('active');
+    elementKey(element);
+    event.preventDefault();
+  }
+}
+
+function keyUp(event) {
+  let code = normalizeKeys(event.code);
+  let element = code && document.getElementById(code);
+  if (element) {
+    element.classList.remove('active');
+    deActiveSystem(code);
+    textarea.focus();
+  }
+}
+
+function changeOneKey(key) {
+  let letter;
+  if (shift) {
+    if (ru && 'rush' in key) {
+      letter = key.rush;
+    } else if (ru && 'ru' in key) {
+      letter = key.ru.toUpperCase();
+    } else if (ru && 'ru_' in key) {
+      letter = key.ru_;
+    } else if (shift && 'sh' in key) {
+      letter = key.sh;
+    } else {
+      letter = key.en.toUpperCase();
+    }
+  } else if (capsLock) {
+    letter = ru && 'ru' in key ? key.ru.toUpperCase() : key.en.toUpperCase();
+  } else {
+    letter = ru && 'ru' in key ? key.ru : key.en;
+  }
+  return letter;
+}
+
+function view() {
+  KEYS.forEach(row => {
+    row.forEach(key => {
+      if (key.form === 'k') {
+        let button = document.querySelector('#' + key.code);
+        button.textContent = changeOneKey(key);
+      }
+    });
+  });
+}
+
+function insert(str, pos, add) {
+  let arr = str.split('');
+  arr.splice(pos, 0, add);
+  return arr.join('');
+}
+
+function del(str, pos, after) {
+  let arr = str.split('');
+  if (after) {
+    arr.splice(pos, 1);
+  } else {
+    pos && arr.splice(pos - 1, 1);
+  }
+  return arr.join('');
+}
+
+function clearTimes(nameKey) {
+  delete times[nameKey];
+}
+
+function systemKey(keyCode) {
+  return (
```

Comment:
That condition can be simplified as `['ShiftLeft', 'ShiftRight', 'AltLeft', 'AltRight', 'ControlLeft', 'ControlRight'].includes(keyCode)`